tools/buildsys/src/cache.rs

```diff
@@ -72,7 +72,15 @@ impl LookasideCache {
             let tmp = PathBuf::from(format!(".{}", name));
 
             // first check the lookaside cache
-            let url = format!("{}/{}/{}/{}", self.lookaside_cache, name, hash, name);
+            let cache_key = format!("{}/{}/{}", name, hash, name);
+            let url = self
+                .lookaside_cache
+                .join(&cache_key)
+                .context(error::UrlJoinSnafu {
+                    path: cache_key,
+                    base: self.lookaside_cache.to_owned(),
+                })?
+                .to_string();
```

Comment:
Was there a reason for this implementations vs the one below other than the extra error management?

```suggestion
           let url = self
                .lookaside_cache
                .join(&name)
                .context(error::UrlJoinSnafu {
                    path: name,
                    base: self.lookaside_cache.to_owned(),
                })?.join(&hash)
                .context(error::UrlJoinSnafu {
                    path: hash,
                    base: self.lookaside_cache.to_owned(),
                })?.join(&name)
                .context(error::UrlJoinSnafu {
                    path: name,
                    base: self.lookaside_cache.to_owned(),
                })?
                .to_string();
```

Reply:
> Was there a reason for this implementations vs the one below other than the extra error management?

Mostly that, yeah. I poked around the url crate a bit and came up with what looks like a more idiomatic approach.